### PR TITLE
Add dns.unpack_name improvements and related tests.

### PR DESCRIPTION
**Pull request by sunhao2014**
This update is totally based on the patch of #146. There are three improvements for `dns.py`.

1. Changes the `unpack_name` from recursive to iterative when there are `ptr`s in the `name` field.
1. Detects pointer loops as soon as possible and raises `UnpackError`.
1. Uses `'.'.join()` to build the return value instead of repeatedly appending to a string to improve the performance.

I checked the logic and run the updated code, it seems good. Also it could pass all tests in `dns.py` module.

### CODE REVIEW

File: dpkt/dns.py [hunk 1]
```diff
@@ -83,28 +83,37 @@ def pack_name(name, off, label_ptrs):
...
 
 def unpack_name(buf, off):
-    name = ''
+    name = []
+    name_length = 0
     saved_off = 0
-    for _ in range(100):  # XXX
+    start_off = off
+    while True:
+        if off > len(buf):
```
Comment (gmacon) [inline dpkt/dns.py]: I've just noticed that this should be `if off >= len(buf)` since we index buf by off on line 93.  Sorry about that.

Comment (brifordwylie) [inline dpkt/dns.py]: @sunhao2014 can you look at/address/comment on @gmacon comment above? After that please go ahead and merge.

Comment (sunhao2014) [inline dpkt/dns.py]: Yes, should be `off >= len(buf)`. Updated and merging. Thanks, @gmacon and @brifordwylie.

**Comment by sunhao2014** [comment]
@brifordwylie, the `continuous-integration/travis-ci/push` build failed, however I'm not sure why. Related log is as follows.
`You have to provide either repo_token in .coveralls.yml, or launch via Travis or CircleCI
ERROR: InvocationError: '/home/travis/build/kbandla/dpkt/.tox/coveralls/bin/coveralls'`
Could you please take a look at this issue?

**Comment by brifordwylie** [comment]
Just did a cherry-pick dd4ec61 for tests/coveralls to pass... (newer tox version requires this env var for Travis).

**Comment by sunhao2014** [comment]
Aha, got it! The branch has been deleted.

**Pull request merged**